sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -315,9 +335,23 @@ private[hive] object HadoopTableReader extends HiveInspectors {
       }
     }
 
+    val partTblObjectInspectorConverter = ObjectInspectorConverters.getConverter(
+      deserializer.getObjectInspector, soi)
+
     // Map each tuple to a row object
     iterator.map { value =>
-      val raw = deserializer.deserialize(value)
+      val raw = convertdeserializer match {
```

Comment:
In general, we'd better not to do the pattern matching within the iterator, and we can do that like:
`
xx match {
  case xxx => iterator.map { ... }
  case yyy => iterator.map { ... }
}
```

For this case, as I shown above, if we passed the converter directly into `fillObject`, I don't think we need the pattern match here.